staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -252,16 +256,100 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 			continue
 		}
 		if evalResult != types.True {
-			if len(rule.Message) != 0 {
-				errs = append(errs, field.Invalid(fldPath, sts.Type, rule.Message))
+			if compiled.MessageExpression != nil {
+				messageExpression, newRemainingBudget, msgErr := evalMessageExpression(ctx, compiled.MessageExpression, rule.MessageExpression, activation, remainingBudget)
+				if msgErr != nil {
+					if errors.Is(msgErr, &runtimeCostError{}) {
+						errs = append(errs, field.InternalError(fldPath, msgErr))
```

Comment:
We previously treat only `evalDetails == nil` as internalError and the rest would be `Invalid` error. Would we keep it consistent?

Reply:
Right, I've added a new `&internalError` type for these situations - fixed